Base/home/anon/tests/run-tests-and-shutdown.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+echo "==== Running Tests on SerenityOS ===="
+
+run() {
+    test_cmd=($*)
+    echo "Running test $idx/$count -- $test_cmd"
+    if $test_cmd {
+        echo "::info file=$test_cmd:: $test_cmd passed!"
+    } else {
+        echo "::error file=$test_cmd:: $test_cmd returned non-zero exit code, check logs!"
+    }
+    idx=$(expr $idx + 1)
+}
+
+# TODO: It'd be nice to have a list+list op (as opposed to nest-on-in-another)
+system_tests=(test-js (test-crypto bigint -t)) #("test-js" "test-web" "test-pthread" "test-crypto" "test-compress" $usr_tests_tests)
+all_tests=($system_tests $(find /usr/Tests -type f | grep -v Kernel | grep -v .inc | shuf))
```

Comment:
This runs *all* current tests, even though most of them are currently broken. How should we deal with this? I think it's a bad idea to just make all builds fail until every single "test" in this directory is repaired; this will take a while.

Reply:
Well, fun fact, printing `::error` doesn't actually fail the build! (who knew.) Last I checked test-js doesn't pass and a few of the others abort instead of forking to check that a particular assertion happens. Seems a #TODO  to make the failing tests fail instead of printing unhappy things at the bottom of the actions run.

Reply:
there's a `test_commons.sh.inc` or somesuch in Shell/Tests, it's just a pass/fail function (for now).

Reply:
> Why exclude the Kernel tests?

At least two of them completely hang -- the futex test and the condition variable test.

> UE fails

Yeah not sure about that one.

There's also whole issue on running tests better in a consistent way here:  #4078 . Hopefully with a reasonable way to run tests we can go back and make a reasonable way to write them.

I also noticed that AK/TestSuite.h doesn't actually return anything from main? It just falls off the end and lets the compiler put the `return 0;`. I was planning to do a subsequent PR to have all the EXPECT and VERIFY macros in that header record to a TestCase::did_fail() thing.

>  I suggest making a blacklist of tests that cannot be run automatically yet.

I mean, this PR has been open a while... Can we close that item to an issue? 😅 

> I'm not sure what you mean by "Seems a #TODO to make the failing tests fail …", could you elaborate?

When a test fails, it will print an `::error` that gets read by actions and is visible in the actions overview. However, it doesn't fail the job. I'm not sure how to make github actions fail the job using that syntax.

Reply:
> I mean, this PR has been open a while... Can we close that item to an issue?

I get what you mean – that's why I suggest to move the problem of failing tests to a different time. For example, blacklisting everything that fails, or only running known-working tests. If we only run two or three tests, manually invoked, that would also circumvent the Shell problems for a while, the PR can get accepted, and there's the opportunity to develop fixes independently. :)

Reply:
That's a good point... would be nice to have things be loud and visible when they don't work properly, but that should be *after* we get to a known good state. I've pared down the set of tests to test-js, test-pthread, test-compress, and test-crypto bigint. These seem to somewhat reliably complete on my machine in non-KVM mode. Still prefer to leave the continue on error  in the run step until #5541 is resolved though.